.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,158 @@
+name: PR
+
+on:
+  pull_request
+
+jobs:
+  python-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          echo "::set-env name=PYTHON_VERSION::$(cat .python-version)"
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install apt dependencies
+        run: sudo apt-get install libsnappy-dev libgconf-2-4
+      - name: Install Pipenv
+        run: pip install pipenv==2018.11.26
+      - name: Cache virtualenv
+        id: cache-virtualenv
+        uses: actions/cache@v1
+        with:
+          path: ~/.local/share/virtualenvs/
+          key: ${{ runner.os }}-${{ env.PYTHON_VERSION }}-virtualenvs-${{ hashFiles('Pipfile.lock') }}
+      - name: Install virtual environment
+        if: steps.cache-virtualenv.outputs.cache-hit != 'true'
+        run: pipenv install --dev
+  node-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '8.16.0'
+      - name: Get yarn cache
+        id: get-yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - name: Cache yarn modules
+        id: cache-yarn
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.get-yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+      - name: Install yarn deps
+        if: steps.cache-yarn.outputs.cache-hit != 'true'
+        run: yarn
+  test-unit:
+    needs: python-dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          echo "::set-env name=PYTHON_VERSION::$(cat .python-version)"
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install apt dependencies
+        run: sudo apt-get install libsnappy-dev libgconf-2-4
+      - name: Write app version
+        run: printf $(git rev-parse HEAD) > .application-version
```

Comment:
I pointed my docker-compose to point at this branch, pulled images and started the services. But when I got to the `/status` endpoint on runner, it doesn't match the current `git rev-parse HEAD`. The version I get is `27309c700371dea3a8e0d414b2b7a7f76f617397`.

Reply:
You need to actually update your local .application-version file prior to doing docker-compose build in order to replicate the steps that travis/actions do when they create a build. Otherwise .application-version will just be whenever you last updated it locally.

I've just done that and it points to this: https://github.com/ONSdigital/eq-questionnaire-runner/commits/5c7bc4ea16d946ae5be89df6f6086603d2971bcb

Reply:
Ah ofc. 👍

Reply:
Hmm, looks like it's incorrectly tagging things at the moment as branch-HEAD. Will investigate.

Reply:
Looks like this is down to how the checkout works. Have updated to use the available github variables and appears to be tagging correctly once again.